google/_modules/gke/cluster.tf

```diff
@@ -49,4 +49,13 @@ resource "google_container_cluster" "current" {
       start_time = var.daily_maintenance_window_start_time
     }
   }
+
+  private_cluster_config {
+    enable_private_nodes    = var.enable_private_nodes
+    enable_private_endpoint = var.enable_private_endpoint
+    master_ipv4_cidr_block  = var.master_cidr_block
+  }
+
+  ip_allocation_policy {}
```

Comment:
```
Step #4 - "terraform plan":       + ip_allocation_policy { # forces replacement
Step #4 - "terraform plan":           + cluster_ipv4_cidr_block       = (known after apply)
Step #4 - "terraform plan":           + cluster_secondary_range_name  = (known after apply)
Step #4 - "terraform plan":           + node_ipv4_cidr_block          = (known after apply)
Step #4 - "terraform plan":           + services_ipv4_cidr_block      = (known after apply)
Step #4 - "terraform plan":           + services_secondary_range_name = (known after apply)
Step #4 - "terraform plan":           + subnetwork_name               = (known after apply)
Step #4 - "terraform plan":         }
```

Reply:
Adding `ip_allocation_policy {}` forces a replacement of the cluster. So this update can not be done without downtime. Even with setting `enable_private_nodes = false` for backwards compatibility, if this attribute is included the cluster gets recreated. Need to find a solution for this.

Reply:
For internal networking, this is required as you need to enforce IP aliasing over routing (This is potentially why it would cause a recreate). Any ideas on how to get around it?

Reply:
Try making this conditional like so:

```hcl
dynamic "ip_allocation_policy" {
  for_each = var.enable_private_nodes ? toset([1]) : null

  content {}
}
```
If `null` doesn't work in the example, try `toset([])`

This should get this blocker out of the way.

Reply:
Also this, @Spazzy757?

Reply:
Again completely missed your last review will hack n this tomorrow